uber/tests/models/test_badge_funcs.py

```diff
@@ -32,52 +37,161 @@ def check_ranges(session):
         assert actual == expected
 
 
-def change_badge(session, attendee, new_type, new_num=0, expected_num=None):
-    session.change_badge(attendee, new_type, new_num)
+def change_badge(session, attendee, new_type, new_num=None, expected_num=None):
+    old_type = attendee.badge_type
+    old_num = attendee.badge_num
+    attendee.badge_num = new_num
+    attendee.badge_type = new_type
```

Comment:
Purely a stylistic suggestion, but I personally prefer tuple assignment:

```python
old_type, old_num = attendee.badge_type, attendee.badge_num
attendee.badge_type, attendee.badge_num = new_type, new_num
```

Your call on which you prefer.